### PR TITLE
Enable embedding lookup/lora_a logic for chunked backend

### PR DESCRIPTION
**Pull request by MogicianWu**
## Motivation

Per https://github.com/sgl-project/sglang/pull/14177, we want to support csgmv backend for lora_a shrink forward/embedding lookup. 

## Modifications

1. Added python/sglang/srt/lora/triton_ops/chunked_embedding_lora_a.py to supported chunked embedding lookup in lora_a operation. Launched #chunks of thread blocks and load embedding along rank dimension.
2. Changed test/registered/lora/test_lora_hf_sgl_logprob_diff.py to use csgmv backend by default.
3. Modified test_shrink_basic inside test/manual/lora/test_chunked_sgmv_backend.py to test lora_a embedding lookup logic as well.
4. Removed lora target module limit in python/sglang/srt/server_args.py. (solved https://github.com/sgl-project/sglang/pull/14796/changes)
5. Added additional triton cache key in python/sglang/srt/lora/triton_ops/chunked_sgmv_expand.py, only use (NUM_SLICES, BLOCK_M) as key will cause collision in test_lora_hf_sgl_logprob_diff.py's test case
```
 [debug] calling _chunked_lora_expand_kernel, cache key:[num_slices=1, BLOCK_M=16] BLOCK_N=64 BLOCK_K=16 OUTPUT_DIM=4096 MAX_RANK=8
 [debug] calling _chunked_lora_expand_kernel, cache key:[num_slices=1, BLOCK_M=16] BLOCK_N=64 BLOCK_K=16 OUTPUT_DIM=32000 MAX_RANK=8
```
6. Compute additional lora batch info metadata for use cases in python/sglang/srt/layers/logits_processor.py where the input activation to lm_head module was pruned 
7. Added guard inside python/sglang/srt/lora/layers.py to detect pruned lm_head shape mistmatch versus precomputed lora batch shape
8. Support multiple calls to lm_head module with varying chunk sizes after lm_head pruning
9. Added a third test to test/registered/lora/test_lora_hf_sgl_logprob_diff.py to test the above test, added debug logs to see first 2 test cases triggered pruning and the third triggered further chunking
```
test_lora_logprob_comparison_basic
[DEBUG logits_processor] lm_head pruning: pruned_states.shape=torch.Size([5, 4096]), skip_chunking=True (enable=False, chunk_size=2048)
[DEBUG logits_processor] lm_head pruning: pruned_states.shape=torch.Size([11, 4096]), skip_chunking=True (enable=False, chunk_size=2048)

test_lora_logprob_comparison_full
[DEBUG logits_processor] lm_head pruning: pruned_states.shape=torch.Size([5, 4096]), skip_chunking=True (enable=False, chunk_size=2048)
[DEBUG logits_processor] lm_head pruning: pruned_states.shape=torch.Size([11, 4096]), skip_chunking=True (enable=False, chunk_size=2048)
[DEBUG logits_processor] lm_head pruning: pruned_states.shape=torch.Size([6, 4096]), skip_chunking=True (enable=False, chunk_size=2048)
[DEBUG logits_processor] lm_head pruning: pruned_states.shape=torch.Size([6, 4096]), skip_chunking=True (enable=False, chunk_size=2048)
[DEBUG logits_processor] lm_head pruning: pruned_states.shape=torch.Size([10, 4096]), skip_chunking=True (enable=False, chunk_size=2048)

test_lora_logprob_comparison_chunked
[DEBUG logits_processor] lm_head multi-pass: pruned_states.shape=torch.Size([5, 4096]), chunk_size=4
[DEBUG logits_processor] pass 1/2: tokens [0:4]
[DEBUG logits_processor] pass 2/2: tokens [4:5]
[DEBUG logits_processor] lm_head multi-pass: pruned_states.shape=torch.Size([11, 4096]), chunk_size=4
[DEBUG logits_processor] pass 1/3: tokens [0:4]
[DEBUG logits_processor] pass 2/3: tokens [4:8]
[DEBUG logits_processor] pass 3/3: tokens [8:11]
[DEBUG logits_processor] lm_head multi-pass: pruned_states.shape=torch.Size([6, 4096]), chunk_size=4
[DEBUG logits_processor] pass 1/2: tokens [0:4]
[DEBUG logits_processor] pass 2/2: tokens [4:6]
[DEBUG logits_processor] lm_head multi-pass: pruned_states.shape=torch.Size([6, 4096]), chunk_size=4
[DEBUG logits_processor] pass 1/2: tokens [0:4]
[DEBUG logits_processor] pass 2/2: tokens [4:6]
[DEBUG logits_processor] lm_head multi-pass: pruned_states.shape=torch.Size([10, 4096]), chunk_size=4
[DEBUG logits_processor] pass 1/3: tokens [0:4]
[DEBUG logits_processor] pass 2/3: tokens [4:8]
[DEBUG logits_processor] pass 3/3: tokens [8:10]
```

## Accuracy Tests

Ran  test/registered/lora/test_lora_hf_sgl_logprob_diff.py test on H100, passed all tests
```
 ~/tmp/sglang/sglang (lora_a_csgmv)]$ pytest test/registered/lora/test_lora_hf_sgl_logprob_diff.py -s -vv --log-cli-level=DEBUG
================================================================================
Overall Statistics
================================================================================

Logprob Differences:
  Prefill:
    Max of max:   1.553154e-02
    Mean of max:  1.090739e-02
    Mean of mean: 3.126751e-03
  Decode:
    Max of max:   3.064442e-02
    Mean of max:  2.019897e-02
    Mean of mean: 3.742597e-03

Logprob Statistics (threshold: 1e-01):
  Overall logprob: 5/5 PASSED
  Prefill logprob: 5/5
  Decode logprob:  5/5

String Statistics:
  Output strings:  5/5

================================================================================
Test completed successfully!
================================================================================
PASSED
```
## Benchmarking and Profiling

Ran sanity check using gsm8k
```
(sglang_env) ~/tmp/sglang/sglang (lora_a_csgmv)]$ python3 -m sglang.test.few_shot_gsm8k --num-questions 200
Downloading from https://raw.githubusercontent.com/openai/grade-school-math/master/grade_school_math/data/test.jsonl to /tmp/test.jsonl
/tmp/test.jsonl: 732kB [00:00, 15.1MB/s]                                                                                                                                                                         
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 200/200 [00:05<00:00, 35.16it/s]
Accuracy: 0.860
Invalid: 0.000
Latency: 5.731 s
Output throughput: 5001.777 token/s
```

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
11. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
12. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
13. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: test/registered/lora/test_chunked_sgmv_backend.py [hunk 1]
```diff
@@ -7,13 +7,18 @@
...
 )
 from sglang.srt.lora.triton_ops.chunked_sgmv_expand import _chunked_lora_expand_kernel
 from sglang.srt.lora.triton_ops.chunked_sgmv_shrink import _chunked_lora_shrink_kernel
 from sglang.srt.lora.utils import LoRABatchInfo
-from sglang.test.lora_utils import reference_sgmv_expand, reference_sgmv_shrink
+from sglang.test.lora_utils import (
+    reference_embedding_lora_a_shrink,
+    reference_sgmv_expand,
+    reference_sgmv_shrink,
+)
```
Comment (Fridge003) [inline test/registered/lora/test_chunked_sgmv_backend.py:25]: Can we register this test as nightly
`register_cuda_ci(est_time=60 suite="nightly-1-gpu", nightly=True)`

File: test/registered/lora/test_chunked_sgmv_backend.py [hunk 2]
```diff
@@ -7,16 +7,24 @@
...
+from sglang.test.ci.ci_register import register_cuda_ci
+from sglang.test.lora_utils import (
+    reference_embedding_lora_a_shrink,
+    reference_sgmv_expand,
+    reference_sgmv_shrink,
+)
 
 CHUNK_SIZE = 16
 
+register_cuda_ci(est_time=60, suite="nightly-1-gpu", nightly=True)
```
Comment (Fridge003) [inline test/registered/lora/test_chunked_sgmv_backend.py:28]: Please move this test to `test/registered` folder

**Comment by yushengsu-thu** [comment]
This PR also resolves the issue in https://github.com/sgl-project/sglang/pull/18649.

**Review by Fridge003 — APPROVED** [review]

**Review by yushengsu-thu — APPROVED** [review]
